### ISSUE TITLE
Understanding Different Network Results 

### ISSUE THREAD

**Issue by janewaitara** [ZackGithui/GameOn#3]
@ZackGithui I understand the Resource in the util shows the different results you get from the network. In this case, Loading would not be a result from the network but a UI state as the app waits for the data. I would like to understand why you added the Loading here?

https://github.com/ZackGithui/GameOn/blob/6177f69777077df2a18cf1247c5b15facde338d7/app/src/main/java/com/example/gameon/util/Resource.kt#L7

**Comment by ZackGithui** [comment]
Confusion with the loading found in states . I've always thought that when you get  data "the Resource loading " will also play part in updating the state one.

**Comment by janewaitara** [comment]
Yes Loading plays a part in updating the UI but it doesn't coming from the Network but from the UI

**Comment by ZackGithui** [comment]
Now I get it. Thank you.

**Comment by janewaitara** [comment]
Check how I did it in this project. I added the loading state in the screen's state. https://github.com/janewaitara/Calories/blob/e019d92498a6e730acd68509cc8ab7f290c61a5c/feature/calories/src/main/java/com/mumbicodes/calories/CaloriesScreenState.kt#L11
